src/launcher/launcher.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/sh
+
+# Copyright The async-profiler authors
+# SPDX-License-Identifier: Apache-2.0
+
+if [[ "$1" == "-v" || "$1" == "--version" ]]; then
+    echo "JFR converter PROFILER_VERSION built on BUILD_DATE"
+    exit 0
+fi
+
+JAVA_CMD=("-Xss2M" "-Dsun.misc.URLClassPath.disableJarChecking")
+
+while [[ $# -gt 0 ]]; do
+    case "$1" in
+        -D*|-X*|-agent*)
+            if [[ ${#1} -gt 2 ]]; then
+                JAVA_CMD+=("$1")
+            fi
+            ;;
+        -J*)
+            JAVA_CMD+=("${1:2}")
+            ;;
+        *)
+            break
+            ;;
+    esac
+    shift
+done
+
+JAVA_CMD+=("-jar" "$0" "$@")
+
+# 1. Try JAVA_HOME
+if [[ -n "$JAVA_HOME" ]] && [[ -x "$JAVA_HOME/bin/java" ]]; then
```

Comment:
AFAIK, `[[` is also a bash extension. Strictly POSIX-compliant shell only supports `[`.